### PR TITLE
fixed filter diverge check

### PR DESCRIPTION
**Pull request by Jackack**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
fixed location calculation for filter diverge check

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [ ] Yes
- [x] No, and this is why: to test in mock roll
- [ ] I need help with writing tests

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Review by TiaSinghania — APPROVED** [review]

File: rb_ws/src/buggy/scripts/auton/rolling_sanity_check.py [hunk 1]
```diff
@@ -85,17 +90,24 @@ def calc_covariance(self):
...
         else:
-            good_seperation = abs(self.filter_location.pose.position.y - self.gps_location.position.y) < 0.5
-            if (not good_seperation):
+            filter_pose = Pose.rospose_to_pose(self.filter_location.pose)
+            filter_loc = World.gps_to_world_pose(filter_pose)
+            rear_antenna_pose = Pose.rospose_to_pose(self.gps_location.pose)
+            rear_antenna_loc = World.gps_to_world_pose(rear_antenna_pose)
+
+            diff = filter_loc - rear_antenna_loc
+            dist = np.sqrt(diff.x ** 2 + diff.y ** 2)
```
Comment (mehulgoel873) [inline rb_ws/src/buggy/scripts/auton/rolling_sanity_check.py:104]: What happens if the z coordinates start to differ, do we not care?

File: rb_ws/src/buggy/scripts/auton/rolling_sanity_check.py [hunk 2]
```diff
@@ -85,17 +90,24 @@ def calc_covariance(self):
...
+            filter_diverge = dist >= self.FILTER_WARN_THRES
+            if filter_diverge:
                 self.warning = 2
 
-            self.filter_gps_status_publisher.publish(good_seperation)
+            # checking less than 1 m
+            self.filter_gps_status_publisher.publish(Bool(not filter_diverge))
```
Comment (mehulgoel873) [inline rb_ws/src/buggy/scripts/auton/rolling_sanity_check.py:110]: We should include some documentation on what true false means for each publisher, but this isn't that big of a deal.

File: rb_ws/src/buggy/scripts/auton/rolling_sanity_check.py (hunk 1 above)
Comment (Jackack) [reply rb_ws/src/buggy/scripts/auton/rolling_sanity_check.py:104]: we don't care

Comment (Jackack) [reply rb_ws/src/buggy/scripts/auton/rolling_sanity_check.py:104]: in fact the z coordinate is thrown out upon conversion to a Pose object

**Pull request merged**